library/system/io_interface.lean

```diff
@@ -40,15 +40,22 @@ class monad_io (m : Type → Type → Type) :=
 (fail     : Π e α, e → m e α)
 (iterate  : Π e α, α → (α → m e (option α)) → m e α)
 -- Primitive Types
+(socket   : Type)
```

Comment:
Can we move that to `monad_io_net_system`?

Reply:
The only thing is that this mirrors the situation with core's `handle : Type`, which is also defined in `monad_io` but is used in the `fs.*` named IO functions.

Reply:
I thought that might be your reason. I think the rationale for `handle` to be there is that it's used in `monad_io_file_system`, in `monad_io_serial` and `monad_io_process` and those classes don't need to depend on each other.

Reply:
Ok sure :)